code/game/machinery/teleporter.dm

```diff
@@ -141,10 +142,48 @@
 	for(var/obj/effect/landmark/minimap_poi/poi in SSholomap.pois)
 		if(poi.z in SSodyssey.scenario_zlevels)
 			possible_pois += poi
+	if(isemptylist(possible_pois))
+		tgui_alert(usr, "Can't find any teleport destinations.", "Teleport Error")
+
+	// actor check for id access
+	if(SSodyssey.scenario && SSodyssey.scenario.actor_accesses && !isemptylist(SSodyssey.scenario.actor_accesses))
+		// check if they have an id at all
+		var/obj/item/card/id/user_id = user.GetIdCard()
+		if(!user_id || !istype(user_id))
+			var/choice = tgui_alert(usr,
```

Comment:
Do not rely on usr, you get the mob as a proc parameter

Reply:
done